app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -44,6 +44,19 @@ def save_ems_infra_inventory(ems, hashes, target = nil)
       return
     end
 
+    # Check if a vm was removed and we still have disk(s)
+    if target.kind_of?(VmOrTemplate) && hashes[:vms].blank?
+      current = hashes[:vm_disks].collect { |disk| URI(disk[:storage][:ems_ref]).path.split('/').last }.uniq
+      if current.blank?
+        storage = nil
+      else
+        vm_storages = ([target.storage] + target.storages).compact.uniq
+        storage = vm_storages.select { |store| !current.include?(store.ems_ref) }
+      end
+
+      target.disconnect_storage storage
+    end
```

Comment:
@blomquisg As you can see in user_remove_vm_finished.yaml we had this step to run this method. It removes storage which makes vm to be always archived. We should check whether the disks are still there and make a decision whether to remove or not. In the above code I get a list of "current" disks and if it is empty I run disconnect_storage as it was or I provide list of storage domains to be removed (if any were not removed).